Search field test ml

gercen8905 opened the pull request:
Guys, it's version without deep refactoring

gercen8905 committed:
Create SearchFieldTest

1. Modify maven.compiler.source and target to version 16
2. Delete default classes from project

gercen8905 committed:
Fix pom.xml issues

czajkaaga reviewed (approved):
it looks nice; simply and clean code :)

dominikbodzioch commented:
Nice use of test annotations! :)

Something happened to the path: src/test/java/**com/epam/qalab**/com/epam/qalab/SearchFieldTest.java
When I checkout to your branch, I can't run test, because it doesn't match with package declaration in file.

Even when I correct package and java version in pom, I still can't run the test.
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
'pom' probably needs more dependencies (I added them in my pull request).

After I complete the missing dependencies, I still have a problem with chromedriver:
"java.lang.IllegalStateException: The driver executable must exist: D:\2022\02_25_amazon_selenium_group1\src\test\resources\chromedriver"
(chromedriver.exe is present under this path)

Unfortunately, I can't run the tests and say anything more about the logic.
I am not sure, but in "tearDown()" method you probably should also call "close" method on the driver (before "quit").

czajkaaga commented:
> After I complete the missing dependencies, I still have a problem with chromedriver: "java.lang.IllegalStateException: The driver executable must exist: D:\2022\02_25_amazon_selenium_group1\src\test\resources\chromedriver" (chromedriver.exe is present under this path)

@dominikbodzioch in my case I used ".exe" at the end of path to my driver file and it works. You can try this:
**\src\test\resources\chromedriver.exe**

gercen8905 committed:
Fix issues after review

1. Delete unnecessary line with sout.
2. Replace 'screen' brackets with ''.
3. Add to pom logging dependency.
4. Fix problem with the path to testing class.

gercen8905 commented:
@dominikbodzioch review changes plz. Suppose that now tests should run.

dominikbodzioch commented:
> @dominikbodzioch review changes plz. Suppose that now tests should run.

"src/test/resources/chromedriver.**exe**" - only this is required to run and pass tests (.exe)

gercen8905 commented:
@dominikbodzioch I think it's because of different operational systems (I have MacOS). When I append .exe to my path test are not executed

dominikbodzioch reviewed (approved):
In a real project, we should adapt to the target environment (driver extension). 
In our conditions, let this variable remain in the "comfortable for the author" version.

gercen8905 commented:
In a real project I think DevOps will run our tests in Jenkins and I heard that there could be some issues with different operation systems (

kristinapeneva commented:
Hello,
The tests run smoothly and are easy to read. I like that you defined correct and incorrect variables on the top and used a separate method search().
In my opinion you may only replace \" in xPath with ', which is shorter. Anyway, very good job!

kristinapeneva reviewed (commented):
I agree with the changes

juliaHrabovska commented on src/test/java/com/epam/qalab/SearchFieldTest.java:
```diff
@@ -0,0 +1,76 @@
+
+
+public class SearchFieldTest {
+    WebDriver driver;
```
All variables can be private
